### DISCUSSION TITLE
onHover event dosen't fired when rendering a GeojsonLayer which have altitude < 0 and pitch=0;

### DISCUSSION THREAD

**Discussion by linze99** [visgl/deck.gl#6674] [Q&A] [answered]
I am using GeojsonLayer to render lines which have some data underground(altitude < 0),
when pitch is small, ex. 0, the onHover event dosen't fired，but if pitch number is larger, it can pick features.
![1](https://user-images.githubusercontent.com/30024479/155087193-b2b171ab-7a35-443f-9e22-8f410f4bb9bd.PNG)
![2](https://user-images.githubusercontent.com/30024479/155087202-e8fdeb57-1d90-4e2b-867e-caf1e08d69fd.PNG)
![3](https://user-images.githubusercontent.com/30024479/155087204-f302f4fa-2434-4c7b-8f8f-bc7960bcb8b2.PNG)

**Comment by padawannn** [comment, marked as answer]
By default, [only the top-level layer's on Event](https://deck.gl/docs/developer-guide/custom-layers/picking#event-propagation) callbacks are invoked so you need to use the [pickMultipleObjects](https://deck.gl/docs/api-reference/core/deck#pickmultipleobjects) method. Maybe [this example](https://github.com/visgl/deck.gl/blob/master/examples/layer-browser/src/app.js) could help you

**Reply by linze99** [reply]
I am using deck.pickObject method to solve this problem, but it has a bad prerformance.
I wonder why the picking system works well when the pitch has a large number,but it fails when i set the map with a smaller pitch?


> By default, [only the top-level layer's on Event](https://deck.gl/docs/developer-guide/custom-layers/picking#event-propagation) callbacks are invoked so you need to use the [pickMultipleObjects](https://deck.gl/docs/api-reference/core/deck#pickmultipleobjects) method. Maybe [this example](https://github.com/visgl/deck.gl/blob/master/examples/layer-browser/src/app.js) could help you

**Reply by Pessimistress** [reply]
If you don't want to pick the other objects that are in front of your GeoJSON line, just set their layer's `pickable` to `false`.

**Reply by linze99** [reply]
There is only on layer in my program, only just one GeoJSON line,some of the fragments have the altitude value < 0 which are under ground.
When the map's pitch has a large number (ex. > 25) the picking system can pick the object well, it is the EXACTLY Expected Behavior.
But when we change the pitch value of the map into a smaller number , we can not pick the object under the ground, 
at the same time, picking other objects which are overground,it works fine.
I want to pick all the object at all pitch values.

> If you don't want to pick the other objects that are in front of your GeoJSON line, just set their layer's `pickable` to `false`.

**Reply by linze99** [reply]
I have a geojson line, which has some sections are under ground, the point data format is :[lng, lat, altitude],when under ground, the altitude is < 0.
this is the issue:
1. when the map's pitch has a large value, i can pick every features ,including the underground features. it is exactly what i expected.
2. when i change the map's pitch into a smaller value,the sections over ground are picked well,but the sections underground cannot be picked. this is not expceted.
3. I use the deck.pickObject method to pick the features underground, but it has a bad performance.
I want to know is there a way to solve this promblem?Thanks a lot.
> If you don't want to pick the other objects that are in front of your GeoJSON line, just set their layer's `pickable` to `false`.

**Reply by Pessimistress** [reply]
Can you create a codepen that reproduces your issue?

**Reply by linze99** [reply]
> Can you create a codepen that reproduces your issue?
here is the source, there are two parts of the line,one is overground,the other part is underground.
please change the pitch using right button of the mouse.when pitch = 0,i cann't  pick the feature that under ground.
[https://codepen.io/linze/pen/RwjBOJb](url)

**Reply by Pessimistress** [reply]
Well you did not mention that you were using MapboxLayer. Your hack to modify the `farZMultiplier` is not effective during picking, so the underground part is in fact clipped when drawn to the picking buffer.

There is no point using the Mapbox context if you are messing with the depth buffer - your layer will not blend correctly with the base map anyways.

**Reply by linze99** [reply]
> 
Thanks a lot! I got it.
do you have another way to render lines and interact with them which have underground parts?

**Reply by Pessimistress** [reply]
Just use regular Deck. https://github.com/visgl/deck.gl/tree/master/examples/get-started/pure-js
